core/variant/variant_parser.cpp

```diff
@@ -1965,7 +1969,7 @@ Error VariantWriter::write(const Variant &p_variant, StoreStringFunc p_store_str
 		} break;
 		case Variant::FLOAT: {
 			String s = rtos_fix(p_variant.operator double(), p_compat);
-			if (s != "inf" && s != "-inf" && s != "nan") {
+			if (s != "INF" && s != "-INF" && s != "NAN" && s != "inf" && s != "inf_neg" && s != "nan") {
```

Comment:
This condition got a bit trickier so a comment about compatibility could exist here, too.

Also, why is `inf_neg` **now** handled, but not` -inf` anymore?

Reply:
This only needs to handle values returned by `rtos_fix`.